### ISSUE TITLE
Can't use a value from a table in it's own conditions

### ISSUE THREAD

**Issue by andyburgessmd** [Budibase/budibase#15009]
**Checklist**
- I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Self
  - Method: docker compose
  - Budibase Version: 3.2.1
  - App Version: 3.2.1

**Describe the bug**
When trying to add a condition to a table you cannot use the values of that table in the statement bindings.
An example usage of this is to set the 'on row click' to a different action based on the count of rows in the table - i.e set a limit.

**To Reproduce**
1. Create a database table with some data
2. Create a table component and set the data source to the database table
3. Go to the conditions tab and add a new condition
4. In either value selector go into the bindings selector. View the bindings and you will see that the table you are in is not shown.

**Expected behavior**
I would expect to see the current table named in the bindings options so that properties of it can be used, for example row count.

**Comment by Stropdasman** [comment]
I agree with this being inconvenient and think this should be fixable. Though you can solve it for now by using a dataprovider and linking the table to it.
